spinalcordtoolbox/utils/shell.py

```diff
@@ -135,8 +135,8 @@ def error(self, message):
             See https://github.com/neuropoly/spinalcordtoolbox/issues/3137.
         """
         # Source: https://stackoverflow.com/a/4042861
-        sys.stderr.write(f'{self.prog}: error: {message}\n\n')
         self.print_help(sys.stderr)
+        sys.stderr.write(f'\n\033[0;01;31m{self.prog}: error: {message}\033[m\n')
```

Comment:
This reminded me of `printv`, so I looked at how we're currently handling the `error` case:

https://github.com/neuropoly/spinalcordtoolbox/blob/c2d2206e21485ab7585af73f4d5b6e895f5fb6e7/spinalcordtoolbox/utils/sys.py#L331

https://github.com/neuropoly/spinalcordtoolbox/blob/c2d2206e21485ab7585af73f4d5b6e895f5fb6e7/spinalcordtoolbox/utils/sys.py#L340

Currently we're using `print` for every case, even for `'error'`. So, would there be any value to refactoring `printv` to switch between `stdout/stderr`. Then, using it here, too?

Reply:
How about that now?

To make it show up on my color scheme I had to bold it, so that's now...inlined for every 'error', and for 'warning' too.

Reply:
Nice! 

Though, does `file` need to be an argument? Doing it this way requires us to backport `stderr` to every instance of `type=['error', 'warning'])`, I think.

I'm wondering if it would be easier to hardcode it as  'error'/'warning' -> stderr, everything else -> stdout.

Reply:
Right, that is a problem.

I don't like `printv` that much; it wraps and combines orthogonal concepts (verbosity, colours and output location). And it replicates features that are in `logging`.

Guessing what the user wants causes more of the same. It would mean users can't `| ` reliably anymore to get the same output (they'd need to `2>&1 | ` instead). We'd have to comb for all uses of `printv(..., type='warning')` and `printv(..., type='error')` and decide if it's safe to split it off like that.

I don't know. This is such a niche detail.

Reply:
Maybe I should just not use printv? I dunno.

Reply:
> Maybe I should just not use printv? I dunno.

Yeah, this sounds like a good call. My suggestion here inadvertently expanded the scope of the change quite a lot.

(Which might be ok if now is the time we want to deal with `printv`, but if not, your original change is the most straightforward.)

Reply:
Come to think of it, the current behaviour is to write errors to stdout. We can think about changing that to be more consistent with the rest of unix but this patch doesn't change any current behaviour, only augments it.